### ISSUE TITLE
32. Keras - binary_crossentropy in code

### ISSUE THREAD

**Issue by letyrodridc** [udacity/aind-issue-reports#203]
I'm opening a new post about this quiz because the issue was reported and further analysis showed that it's referencing to both categorical and binary crossentropy. 

In the code of the quiz says:
![image](https://user-images.githubusercontent.com/9123307/34020440-00a1f364-e113-11e7-9f91-faddaba92fb3.png)

1) The comment indicates to use `binary_crossentropy` and the solution uses `categorical_crossentropy`.

2) Also the text was corrected in a previous post https://github.com/udacity/aind-issue-reports/issues/178, but it's implicit referencing binary cross-entropy and explicit to categorical_crossentropy: 

> We'll specify the loss function to be categorical_crossentropy which can be used when there are only two classes

3) The Quiz pass using both binary_crossentropy or categorical_crossentropy. 

```
xor.compile(loss="binary_crossentropy", optimizer="adam", metrics = ['accuracy'])

# Uncomment this line to print the model architecture
# xor.summary()

# Fitting the model
# Hint: This next line is where you can change the number of epochs, it's set to 10 now.
history = xor.fit(X, y, nb_epoch=100, verbose=0)
```

4) The solution uses categorical_crossentropy.

Link to the lesson: https://classroom.udacity.com/nanodegrees/nd889/parts/16cf5df5-73f0-4afa-93a9-de5974257236/modules/6124bd95-dec2-44f9-bf3b-498ea57699c7/lessons/47f6c25c-7749-4a02-b807-7a5b37f362e8/concepts/7c3d9b4c-206d-4430-9459-c78357181d9a

**Comment by letyrodridc** [comment]
Also, student reports:
>Then epochs don’t match and the solution has one-hot encoding but never mentions this. Could someone look at this because it seems completely wrong.

**Comment by letyrodridc** [comment]
Link to the lesson: https://classroom.udacity.com/nanodegrees/nd889/parts/16cf5df5-73f0-4afa-93a9-de5974257236/modules/6124bd95-dec2-44f9-bf3b-498ea57699c7/lessons/47f6c25c-7749-4a02-b807-7a5b37f362e8/concepts/7c3d9b4c-206d-4430-9459-c78357181d9a

**Comment by letyrodridc** [comment]
Also the text was corrected in a previous post, but it's referencing binary cross-entropy: 
> We'll specify the loss function to be categorical_crossentropy which can be used when there are only two classes

The Quiz pass using both binary_crossentropy or categorical_crossentropy. 

```
xor.compile(loss="binary_crossentropy", optimizer="adam", metrics = ['accuracy'])

# Uncomment this line to print the model architecture
# xor.summary()

# Fitting the model
# Hint: This next line is where you can change the number of epochs, it's set to 10 now.
history = xor.fit(X, y, nb_epoch=100, verbose=0)
```

**Comment by letyrodridc** [comment]
Another: The instructions require to use `tanh` activation but the solution uses `softmax`.

**Comment by cgearhart** [comment]
Internal ticket #J1915
